pkg/reconciler/labeler/v1/accessors.go

```diff
@@ -34,20 +36,29 @@ type accessor interface {
 
 // revision is an implementation of accessor for Revisions
 type revision struct {
-	r *Reconciler
+	client         clientset.Interface
+	revisionLister listers.RevisionLister
 }
 
 // revision implements accessor
 var _ accessor = (*revision)(nil)
 
+// MakeRevisionAccessor is a factory function to make a new revision accessor
+func MakeRevisionAccessor(client clientset.Interface, lister listers.RevisionLister) *revision {
```

Comment:
Returning private types from public methods can get weird. Should this rather return `accessor` and should `accessor` then be public?

Reply:
No because the dependant code uses this as a revision to get a revisionLister to get concrete types (and only uses it as Accessor when applying the RouteLabel). This is a sort of elaborate way to get the listers from the Reconciler without a circular dep on Reconciler.

Reply:
made public